modules/subscriptions/subscriptions/subscriptions.tsx

```diff
@@ -1,65 +1,85 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { useTranslation } from "react-i18next";
 import { GlobalCore } from "@/core/module/module.types";
 import prices from "./pricing-config.json";
 import Service from "@/core/module/service.factory";
 import styles from "./styles/subscriptions.module.css";
 import styled from "styled-components";
+import { SubscriptionPriceData } from "@/core/module/core.types";
 
 type Price = (typeof prices)[0];
 
-const PriceCard = (props: { item: Price }) => {
+const PriceCard = (props: { item: Price; index: number }) => {
   const { t } = useTranslation();
   const [isLoading, setIsLoading] = useState(false);
   const {
     title,
-    currency,
-    amount,
     interval,
+    amount,
+    currency,
     features,
     buttonText,
     buttonAction,
   } = props.item;
+  const [priceData, setPriceData] = useState<SubscriptionPriceData>();
```

Comment:
We are using null, lets the typescript to be aware of ` useState<SubscriptionPriceData | null>(null);`